src/util/cmdlineargs.cpp

```diff
@@ -36,89 +38,197 @@ CmdlineArgs::CmdlineArgs()
 }
 
 namespace {
-    bool parseLogLevel(
-        QLatin1String logLevel,
-        mixxx::LogLevel *pLogLevel) {
-        if (logLevel == QLatin1String("trace")) {
-            *pLogLevel = mixxx::LogLevel::Trace;
-        } else if (logLevel == QLatin1String("debug")) {
-            *pLogLevel = mixxx::LogLevel::Debug;
-        } else if (logLevel == QLatin1String("info")) {
-            *pLogLevel = mixxx::LogLevel::Info;
-        } else if (logLevel == QLatin1String("warning")) {
-            *pLogLevel = mixxx::LogLevel::Warning;
-        } else if (logLevel == QLatin1String("critical")) {
-            *pLogLevel = mixxx::LogLevel::Critical;
-        } else {
-            return false;
-        }
-        return true;
+bool parseLogLevel(
+        const QString& logLevel,
+        mixxx::LogLevel* pLogLevel) {
+    if (logLevel.compare(QLatin1String("trace"), Qt::CaseInsensitive) == 0) {
+        *pLogLevel = mixxx::LogLevel::Trace;
+    } else if (logLevel.compare(QLatin1String("debug"), Qt::CaseInsensitive) == 0) {
+        *pLogLevel = mixxx::LogLevel::Debug;
+    } else if (logLevel.compare(QLatin1String("info"), Qt::CaseInsensitive) == 0) {
+        *pLogLevel = mixxx::LogLevel::Info;
+    } else if (logLevel.compare(QLatin1String("warning"), Qt::CaseInsensitive) == 0) {
+        *pLogLevel = mixxx::LogLevel::Warning;
+    } else if (logLevel.compare(QLatin1String("critical"), Qt::CaseInsensitive) == 0) {
+        *pLogLevel = mixxx::LogLevel::Critical;
+    } else {
+        return false;
     }
+    return true;
 }
+} // namespace
+
+bool CmdlineArgs::parse(const QStringList& arguments) {
+    QCommandLineParser parser;
+    parser.setApplicationDescription(QCoreApplication::translate("main",
+            "Mixxx is an open source DJ software. For more information, see "
+            "https://manual.mixxx.org/2.3/chapters/appendix.html#command-line-options).\n"
+            "CamelCase arguments are deprecated and will be removed in 2.5"));
```

Comment:
```suggestion
            "https://manual.mixxx.org/2.4/chapters/appendix.html#command-line-options"));
```
Please make a PR for the manual too.